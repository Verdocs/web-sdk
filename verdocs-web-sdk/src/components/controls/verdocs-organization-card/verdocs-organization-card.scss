@import '../../../theme.scss';

verdocs-organization-card {
  font-family: $primary-font;
  width: 242px;
  position: relative;
  display: flex;
  flex-direction: row;

  &:hover > .popup {
    display: flex;
    background: #ffffff;
    color: $medium-bg;
  }

  .icon {
    flex: 0;
    margin: 0 7px 0 0;

    svg {
      fill: $icon-color;
      width: 24px;
      margin-bottom: -5px;
    }
  }

  .popup {
    position: absolute;
    top: -1px;
    left: -4px;
    display: none;
    border: 1px solid $dark-bg;
    border-radius: 10px;
    width: 326px;
    height: 220px;
    flex-direction: column;
    flex-wrap: nowrap;
    align-items: center;
    justify-content: space-evenly;
    font-family: sans-serif;
  }

  .popup-header {
    display: flex;
    flex-direction: row;
    align-items: center;
    width: 242px;
    height: 69px;
    justify-content: space-between;
    font-size: 21px;
    color: $secondary-color;
  }

  .popup-body {
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    width: 256.2px;
    height: 95px;
    justify-content: space-evenly;
    font-size: 13px;
    text-transform: uppercase;

    span {
      display: block;
      flex: 2 80px;
      margin-left: 7px;

      &:last-child {
        margin-top: 5px;
      }
    }
  }

  .status {
    width: 29px;
    height: 29px;
    background-color: $icon-color;
    border-radius: 50%;
  }
}
