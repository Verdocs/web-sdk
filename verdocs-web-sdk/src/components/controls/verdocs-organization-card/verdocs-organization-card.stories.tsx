import {Meta} from '@storybook/web-components';
import {html} from 'lit-html';
import {IOrganization} from '@verdocs/js-sdk/Organizations/Types';

const MockOrganization = {
  id: '21643fd6-4097-47c0-9780-62dc0c01ddce',
  name: 'Verdocs, LLC',
  url: 'https://verdocs.com/',
  address: null,
  phone: null,
  business_name: null,
  is_business: true,
  address2: null,
  contact_email: null,
  timezone: null,
  envelope_responsible: false,
  created_at: '2022-12-21 20:16:50.262113+00',
} as IOrganization;

export default {
  title: 'Controls/Organization Card',
  component: 'verdocs-organization-card',
  args: {
    organization: MockOrganization,
  },
  argTypes: {},
} as Meta;

export const OrganizationCard = ({organization}) => html`<verdocs-organization-card .organization=${organization} />`;
