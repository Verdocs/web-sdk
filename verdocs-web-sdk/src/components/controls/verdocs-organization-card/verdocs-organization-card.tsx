import {format} from 'date-fns';
import {Component, Prop, h, Host} from '@stencil/core';
import {IOrganization} from '@verdocs/js-sdk';

// TODO: See what was done in Storybook. Move all SVG icons to a common file that both projects can share
//  and reduce duplication between components for re-used icons. Also bring icons into Figma for reuse there.

const UserIcon =
  '<svg xmlns="http://www.w3.org/2000/svg" height="24px" viewBox="0 0 24 24" width="24px"><rect fill="none" height="24" width="24"/><g><path d="M4,13c1.1,0,2-0.9,2-2c0-1.1-0.9-2-2-2s-2,0.9-2,2C2,12.1,2.9,13,4,13z M5.13,14.1C4.76,14.04,4.39,14,4,14 c-0.99,0-1.93,0.21-2.78,0.58C0.48,14.9,0,15.62,0,16.43V18l4.5,0v-1.61C4.5,15.56,4.73,14.78,5.13,14.1z M20,13c1.1,0,2-0.9,2-2 c0-1.1-0.9-2-2-2s-2,0.9-2,2C18,12.1,18.9,13,20,13z M24,16.43c0-0.81-0.48-1.53-1.22-1.85C21.93,14.21,20.99,14,20,14 c-0.39,0-0.76,0.04-1.13,0.1c0.4,0.68,0.63,1.46,0.63,2.29V18l4.5,0V16.43z M16.24,13.65c-1.17-0.52-2.61-0.9-4.24-0.9 c-1.63,0-3.07,0.39-4.24,0.9C6.68,14.13,6,15.21,6,16.39V18h12v-1.61C18,15.21,17.32,14.13,16.24,13.65z M8.07,16 c0.09-0.23,0.13-0.39,0.91-0.69c0.97-0.38,1.99-0.56,3.02-0.56s2.05,0.18,3.02,0.56c0.77,0.3,0.81,0.46,0.91,0.69H8.07z M12,8 c0.55,0,1,0.45,1,1s-0.45,1-1,1s-1-0.45-1-1S11.45,8,12,8 M12,6c-1.66,0-3,1.34-3,3c0,1.66,1.34,3,3,3s3-1.34,3-3 C15,7.34,13.66,6,12,6L12,6z"/></g></svg>';

const BusinessIcon =
  '<svg xmlns="http://www.w3.org/2000/svg"  focusable="false" aria-hidden="true" height="24px" viewBox="0 0 24 24" width="24px"><path d="M12 7V3H2v18h20V7H12zM6 19H4v-2h2v2zm0-4H4v-2h2v2zm0-4H4V9h2v2zm0-4H4V5h2v2zm4 12H8v-2h2v2zm0-4H8v-2h2v2zm0-4H8V9h2v2zm0-4H8V5h2v2zm10 12h-8v-2h2v-2h-2v-2h2v-2h-2V9h8v10zm-2-8h-2v2h2v-2zm0 4h-2v2h2v-2z"></path></svg>';

/**
 * Display a small summary card describing an organization
 */
@Component({
  tag: 'verdocs-organization-card',
  styleUrl: 'verdocs-organization-card.scss',
})
export class VerdocsOrganizationCard {
  /**
   * The organization to display
   */
  @Prop() organization: IOrganization;

  render() {
    console.log('Rendering org card', this.organization.thumbnail_url, this.organization);
    const fallbackIcon = this.organization?.is_business ? BusinessIcon : UserIcon;

    return (
      <Host>
        {this.organization.thumbnail_url ? <img class="icon" src={this.organization.thumbnail_url} alt="Logo" /> : <span class="icon" innerHTML={fallbackIcon} />}

        <span class="content">{this.organization?.name}</span>

        <div class="popup">
          {this.organization.full_logo_url ? (
            <div class="popup-header-with-logo">
              <img src={this.organization.full_logo_url} alt="Logo" />
              <div class="title">{this.organization?.name}</div>
            </div>
          ) : (
            <div class="popup-header">
              <span class="status">
                <span class="icon" innerHTML={fallbackIcon} />
              </span>
              <span class="title">{this.organization?.name}</span>
            </div>
          )}

          <div class="popup-body">
            <div class="field">
              <strong>Joined</strong>
              {format(new Date(this.organization?.created_at || undefined), 'LLL d, Y')}
            </div>
            {this.organization.url && (
              <div class="field">
                <strong>Website</strong>
                <a href={this.organization?.url} target="_blank" rel="nofollow">
                  {this.organization?.url}
                </a>
              </div>
            )}
          </div>
        </div>
      </Host>
    );
  }
}
