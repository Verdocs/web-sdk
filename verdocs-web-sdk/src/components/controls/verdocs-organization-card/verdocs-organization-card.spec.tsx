import {IOrganization} from '@verdocs/js-sdk';
import {newSpecPage} from '@stencil/core/testing';
import {VerdocsOrganizationCard} from './verdocs-organization-card';
import {h} from '@stencil/core';

describe('verdocs-organization-card', () => {
  const baseOrg: IOrganization = {
    address: '',
    address2: '',
    contact_email: '',
    parent_id: '',
    phone: '',
    updated_at: '',
    id: 'org1',
    name: 'Test Org',
    thumbnail_url: '',
    full_logo_url: '',
    created_at: '2021-01-01T00:00:00Z',
  };

  it('renders with minimal organization data', async () => {
    const page = await newSpecPage({
      components: [VerdocsOrganizationCard],
      template: () => <verdocs-organization-card organization={{...baseOrg}} />,
    });
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });

  it('renders with thumbnail and full logo', async () => {
    const org = {
      ...baseOrg,
      thumbnail_url: 'https://example.com/thumb.png',
      full_logo_url: 'https://example.com/logo.png',
    };
    const page = await newSpecPage({
      components: [VerdocsOrganizationCard],
      template: () => <verdocs-organization-card organization={org} />,
    });
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });

  it('renders with only name', async () => {
    const org: IOrganization = {
      address: '',
      address2: '',
      contact_email: '',
      parent_id: '',
      phone: '',
      updated_at: '',
      id: 'org2',
      name: 'Name Only',
      thumbnail_url: '',
      full_logo_url: '',
      created_at: '2022-02-02T00:00:00Z',
    };
    const page = await newSpecPage({
      components: [VerdocsOrganizationCard],
      template: () => <verdocs-organization-card organization={org} />,
    });
    await page.waitForChanges();
    expect(page.root).toMatchSnapshot();
  });
});
