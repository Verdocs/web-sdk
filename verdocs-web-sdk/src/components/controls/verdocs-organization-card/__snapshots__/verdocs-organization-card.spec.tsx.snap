// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verdocs-organization-card renders with minimal organization data 1`] = `
<verdocs-organization-card id="verdocs-org-card-org1">
  <span class="icon">
    <svg aria-hidden="true" focusable="false" height="24px" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
      <path d="M12 7V3H2v18h20V7H12zM6 19H4v-2h2v2zm0-4H4v-2h2v2zm0-4H4V9h2v2zm0-4H4V5h2v2zm4 12H8v-2h2v2zm0-4H8v-2h2v2zm0-4H8V9h2v2zm0-4H8V5h2v2zm10 12h-8v-2h2v-2h-2v-2h2v-2h-2V9h8v10zm-2-8h-2v2h2v-2zm0 4h-2v2h2v-2z"></path>
    </svg>
  </span>
  <span class="content">
    Test Org
  </span>
  <verdocs-portal anchor="verdocs-org-card-org1">
    <div class="verdocs-org-card-popup" style="display: none;">
      <div class="popup-header">
        <span class="status">
          <span class="icon">
            <svg aria-hidden="true" focusable="false" height="24px" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
              <path d="M12 7V3H2v18h20V7H12zM6 19H4v-2h2v2zm0-4H4v-2h2v2zm0-4H4V9h2v2zm0-4H4V5h2v2zm4 12H8v-2h2v2zm0-4H8v-2h2v2zm0-4H8V9h2v2zm0-4H8V5h2v2zm10 12h-8v-2h2v-2h-2v-2h2v-2h-2V9h8v10zm-2-8h-2v2h2v-2zm0 4h-2v2h2v-2z"></path>
            </svg>
          </span>
        </span>
        <span class="title">
          Test Org
        </span>
      </div>
      <div class="popup-body">
        <div class="field">
          <strong>
            Joined
          </strong>
          Jan 1, 2021
        </div>
      </div>
    </div>
  </verdocs-portal>
</verdocs-organization-card>
`;

exports[`verdocs-organization-card renders with only name 1`] = `
<verdocs-organization-card id="verdocs-org-card-org2">
  <span class="icon">
    <svg aria-hidden="true" focusable="false" height="24px" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
      <path d="M12 7V3H2v18h20V7H12zM6 19H4v-2h2v2zm0-4H4v-2h2v2zm0-4H4V9h2v2zm0-4H4V5h2v2zm4 12H8v-2h2v2zm0-4H8v-2h2v2zm0-4H8V9h2v2zm0-4H8V5h2v2zm10 12h-8v-2h2v-2h-2v-2h2v-2h-2V9h8v10zm-2-8h-2v2h2v-2zm0 4h-2v2h2v-2z"></path>
    </svg>
  </span>
  <span class="content">
    Name Only
  </span>
  <verdocs-portal anchor="verdocs-org-card-org2">
    <div class="verdocs-org-card-popup" style="display: none;">
      <div class="popup-header">
        <span class="status">
          <span class="icon">
            <svg aria-hidden="true" focusable="false" height="24px" viewBox="0 0 24 24" width="24px" xmlns="http://www.w3.org/2000/svg">
              <path d="M12 7V3H2v18h20V7H12zM6 19H4v-2h2v2zm0-4H4v-2h2v2zm0-4H4V9h2v2zm0-4H4V5h2v2zm4 12H8v-2h2v2zm0-4H8v-2h2v2zm0-4H8V9h2v2zm0-4H8V5h2v2zm10 12h-8v-2h2v-2h-2v-2h2v-2h-2V9h8v10zm-2-8h-2v2h2v-2zm0 4h-2v2h2v-2z"></path>
            </svg>
          </span>
        </span>
        <span class="title">
          Name Only
        </span>
      </div>
      <div class="popup-body">
        <div class="field">
          <strong>
            Joined
          </strong>
          Feb 2, 2022
        </div>
      </div>
    </div>
  </verdocs-portal>
</verdocs-organization-card>
`;

exports[`verdocs-organization-card renders with thumbnail and full logo 1`] = `
<verdocs-organization-card id="verdocs-org-card-org1">
  <img alt="Logo" class="icon" src="https://example.com/thumb.png">
  <span class="content">
    Test Org
  </span>
  <verdocs-portal anchor="verdocs-org-card-org1">
    <div class="verdocs-org-card-popup" style="display: none;">
      <div class="popup-header-with-logo">
        <img alt="Logo" src="https://example.com/logo.png">
        <div class="title">
          Test Org
        </div>
      </div>
      <div class="popup-body">
        <div class="field">
          <strong>
            Joined
          </strong>
          Jan 1, 2021
        </div>
      </div>
    </div>
  </verdocs-portal>
</verdocs-organization-card>
`;
